### PR TITLE
refactor: extract methods from main cli 

### PR DESCRIPTION
**Pull request by lionel-nj**
Relates to #734 

**Summary:**

This PR provides support to extract intermediate validation methods. The is needed for #933.

**Expected behavior:** 

Same behavior as before method extraction.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- ~[] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/cli/Main.java [hunk 1]
```diff
@@ -157,6 +136,79 @@ public static void main(String[] argv) {
...
+            new DefaultValidatorProvider(validationContext, validatorLoader),
+            noticeContainer);
+    return feedContainer;
+  }
+
+  /**
+   * Performs parsing and sanity checks on CLI arguments.
+   *
+   * @param argv the CLI arguments
+   * @return the {@code Argument} generated after parsing the command line
```
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/cli/Main.java:188]: Should this be `Arguments`?

File: main/src/main/java/org/mobilitydata/gtfsvalidator/cli/Main.java [hunk 2]
```diff
@@ -157,6 +136,79 @@ public static void main(String[] argv) {
...
+   * @param validatorLoader the {@code ValidatorLoader} used in the process
+   * @param feedLoader the {@code GtfsFeedLoader} used in the process
+   * @param noticeContainer the {@code NoticeContainer} that will contain {@code Notice}s related to
+   *     the GTFS feed
+   * @param gtfsInput the source of data
+   * @param validationContext the {@code ValidationContext} do be used during validation
+   * @return the {@code GtfsFeedContainer} used in the validation process
+   * @throws InterruptedException if validation process was interrupted
+   */
+  public static GtfsFeedContainer validate(
```
Comment (barbeau) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/cli/Main.java:168]: Should this be called `loadAndValidate()` too?

**Review by barbeau — COMMENTED** [review]
Thanks @lionel-nj, I think this looks good. I think it definitely makes the `main()` easier to read. A few comments in-line.

So for code using gtfs-validator as a library now, the code calling the library would look a lot like the `main()` method, but with that code choosing how to handle exceptions, etc., right? So for example instead of doing `System.exit(1)` when `gtfsInput == null` it might just log the error (or respond with an error in case of a hosted web service) and return.

**Comment by lionel-nj** [comment]
Thanks for reviewing @barbeau.

> So for code using gtfs-validator as a library now, the code calling the library would look a lot like the main() method, but with that code choosing how to handle exceptions, etc., right? So for example instead of doing System.exit(1) when gtfsInput == null it might just log the error (or respond with an error in case of a hosted web service) and return.

Exactly, that is the goal.

**Review by barbeau — APPROVED** [review]
👍

**Pull request merged**